cmd/panfs_test.go

```diff
@@ -363,3 +370,37 @@ func TestPANFSHealObjects(t *testing.T) {
 		t.Fatalf("Heal Object should return NotImplemented error ")
 	}
 }
+
+// TestCheckForS3Prefix
+func TestDotS3PrefixCheck(t *testing.T) {
+	testCases := []struct {
+		input         []string
+		shouldPass    bool
+		expectedError error
+	}{
+		{[]string{"valid", ""}, true, nil},
+		{[]string{}, true, nil},
+		{[]string{"file", ".s3file"}, true, nil},
+		{[]string{"file", ".s3/file"}, false, PanFSS3InvalidName{}},
+		{[]string{"file", ".s3/file", ".s3/nextfile"}, false, PanFSS3InvalidName{}},
+	}
+
+	for _, tc := range testCases {
+		err := dotS3PrefixCheck(tc.input...)
+		if err != nil && !tc.shouldPass {
+			if !errors.Is(err, tc.expectedError) {
```

Comment:
Maybe it makes sense to log `tc.input` in those calls to `t.Fatalf()` to make things obvious in case of problems?